presto-record-decoder/src/main/java/io/prestosql/decoder/json/AbstractDateTimeJsonValueProvider.java

```diff
@@ -66,7 +67,10 @@ public final long getLong()
         if (type.equals(DATE)) {
             return TimeUnit.MILLISECONDS.toDays(millis);
         }
-        if (type.equals(TIMESTAMP) || type.equals(TIME)) {
+        if (type.equals(TIME)) {
+            return millis * PICOSECONDS_PER_MILLISECOND;
```

Comment:
can overflow?

also, depending on what getMillis() above does, this needs to round to `[0, PICOS_PER_DAY)`

Reply:
It shouldn't. If it's a `TIME`, the value should be between `[0, MILLIS_PER_DAY]`. See the check above: `if (millis < 0 || millis >= TimeUnit.DAYS.toMillis(1)) { ... }`